### ISSUE TITLE
Multi-Result queries may mix ResultSets with update counts. jOOQ doesn't correctly check both in fetchMany()

### ISSUE THREAD

**Issue by lukaseder** [jOOQ/jOOQ#2364]
It appears that the current implementation of `fetchMany()` is flawed, when update counts are mixed with `ResultSets`. A more correct implementation seems to be this:

```java
 for (int index = 0 ; ; index ++){
        if (callable.getMoreResults()){
            ResultSet results = callable.getResultSet();
            int count = 0 ;
            while (results.next()){
                count++;
            }
            if (index == 1){
                partitions = count;
            }
        } else if (callable.getUpdateCount() == -1){
            break ;
        }
    }
```

See also this Stack Overflow answer:
http://stackoverflow.com/a/15607411/521799

**Comment by lukaseder** [comment]
This defect has been fixed somewhere along the way with another defect / feature request.

**Comment by lukaseder** [comment]
This fix depends on #4502 and will thus not be merged to older versions.
